packages/performance/ios/Plugin/FirebasePerformancePlugin.swift

```diff
@@ -80,4 +85,141 @@ public class FirebasePerformancePlugin: CAPPlugin {
+            return
+        }
+        guard let attribute = call.getString("attribute") else {
+            call.reject(errorAttributeMissing)
+            return
+        }
+        let trace = implementation?.getTraceByName(traceName)
+        guard trace != nil else {
+            call.reject(errorTraceNotFound)
+            return
+        }
+        var result = JSObject()
+        result["value"] = FirebasePerformance.getAttribute(trace!, attribute) ?? NSNull()
+        call.resolve(result)
+    }
+
+    @objc func getAttributes(_ call: CAPPluginCall) {
+        guard let traceName = call.getString("traceName") else {
+            call.reject(errorTraceNameMissing)
+            return
+        }
+        let trace = implementation?.getTraceByName(traceName)
+        guard trace != nil else {
+            call.reject(errorTraceNotFound)
+            return
+        }
+        call.resolve(["result": FirebasePerformance.getAttributes(trace!)])
+    }
+
+    @objc func removeAttribute(_ call: CAPPluginCall) {
+        guard let traceName = call.getString("traceName") else {
+            call.reject(errorTraceNameMissing)
+            return
+        }
+        guard let attribute = call.getString("attribute") else {
+            call.reject(errorAttributeMissing)
+            return
+        }
+        let trace = implementation?.getTraceByName(traceName)
+        guard trace != nil else {
+            call.reject(errorTraceNotFound)
+            return
+        }
+        FirebasePerformance.removeAttribute(trace!, attribute)
+        call.resolve()
+    }
+
+    @objc func putMetric(_ call: CAPPluginCall) {
+        guard let traceName = call.getString("traceName") else {
+            call.reject(errorTraceNameMissing)
+            return
+        }
+        guard let metricName = call.getString("metricName") else {
+            call.reject(errorMetricNameMissing)
+            return
+        }
+        guard let num = call.getDouble("num") else {
+            call.reject(errorNumMissing)
+            return
+        }
+        let trace = implementation?.getTraceByName(traceName)
+        guard trace != nil else {
+            call.reject(errorTraceNotFound)
+            return
+        }
+        FirebasePerformance.putMetric(trace!, metricName, num)
+        call.resolve()
+    }
+
+    @objc func getMetric(_ call: CAPPluginCall) {
+        guard let traceName = call.getString("traceName") else {
+            call.reject(errorTraceNameMissing)
+            return
+        }
+        guard let metricName = call.getString("metricName") else {
+            call.reject(errorMetricNameMissing)
+            return
+        }
+        let trace = implementation?.getTraceByName(traceName)
+        guard trace != nil else {
+            call.reject(errorTraceNotFound)
+            return
+        }
+        let value = FirebasePerformance.getMetric(trace!, metricName)
+        call.resolve(["value": value == 0 ? NSNull() : value])
```

Comment:
```suggestion
        call.resolve(["value": value])
```

We should not change the behavior of the Firebase SDKs. These plugins are just wrappers.